cpp/daal/src/algorithms/dtrees/forest/classification/oneapi/df_classification_predict_dense_oneapi_impl.i

```diff
@@ -286,11 +288,32 @@ services::Status PredictKernelOneAPI<algorithmFPType, method>::predictByAllTrees
         if (weighted)
         {
             const double * probas = pModel->getProbas(iTree);
-            context.copy(probasArr, iTree * maxTreeSize * _nClasses, (void *)probas, treeSize * _nClasses, 0, treeSize * _nClasses, status);
-            DAAL_CHECK_STATUS_VAR(status);
+
+            if (sizeof(algorithmFPType) == sizeof(double))
+            {
+                services::internal::tmemcpy<double, sse2>(reinterpret_cast<double *>(probasArrHost.get()) + iTree * maxTreeSize * _nClasses, probas,
+                                                          treeSize * _nClasses);
+            }
+            else
+            {
+                algorithmFPType * dst_ptr = probasArrHost.get() + iTree * maxTreeSize * _nClasses;
+
+                PRAGMA_IVDEP
+                PRAGMA_VECTOR_ALWAYS
+                for (size_t i = 0; i < treeSize * _nClasses; i++)
+                {
+                    dst_ptr[i] = static_cast<algorithmFPType>(probas[i]);
+                }
+            }
```

Comment:
I believe we can leave only one branch here:
```c++
algorithmFPType * dst_ptr = probasArrHost.get() + iTree * maxTreeSize * _nClasses;                
PRAGMA_IVDEP                
PRAGMA_VECTOR_ALWAYS                
for (size_t i = 0; i &lt; treeSize * _nClasses; i++)                
{                    
   dst_ptr[i] = static_cast&lt;algorithmFPType&gt;(probas[i]);                
}
```

Reply:
Yes, it can be combined here. I wanted to use optimized copy function, but I checked that it provides the same perf as second variant, therefore used second one for both cases.